### PR TITLE
chore(all): Remove bindMethods in favor of Class properties

### LINKED ISSUE #294: Remove bindMethods in favor of Class properties
Class properties are on their way to making it into the spec. Moving to them will also align us better with the greater React community since many projects have adopted this syntax.  Example of the change is below:
### Before
```jsx
class SomeComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
    };
    bindMethods(this, ['handleClick']);
  }
  handleClick() {}
}
```
### After
```jsx
class SomeComponent extends Component {
  state = {
    isLoading: false,
  };
  handleClick = () => {}
}
```
[TC39 Proposal](https://github.com/tc39/proposal-class-fields)
[Stage Explanations](https://tc39.github.io/process-document/)

### PR DESCRIPTION
**Pull request by chalettu**
fixes #294

### CODE REVIEW

**Comment by priley86** [comment]
yea - we are using those downstream too (directly from pf-react)...so will just make a note for the next major. 👍

**Comment by chalettu** [comment]
Updated PR with a console.warn message reflecting that bindMethods is now being deprecated.

File: src/common/helpers.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ import React from 'react';
 
 /** Equivalent to calling `this.someMethod = this.someMethod.bind(this)` for every method name in the methods array. */
 export const bindMethods = (context, methods) => {
+  // eslint-disable-next-line no-console
+  console.warn(`
+              bindMethods usage is being deprecated in favor of class methods.  
+              bindMethods will be deprecated in next major release
+              `);
```
Comment (jeff-phillips-18) [inline src/common/helpers.js]: Slight text update:
`is deprecated`
`will be removed`

**Comment by chalettu** [comment]
@jeff-phillips-18 , I updated the language in the console message.  Let me know if you think we need any more changes.  Thanks in advance

**Comment by priley86** [comment]
@chalettu did you rebase the inline edit table?

**Comment by priley86** [comment]
Thanks a lot! I will do a quick sanity check in the morning but this looks good ;)

**Comment by jeff-phillips-18** [comment]
I believe there is still a call to `bindMethods` in  `src/components/Wizard/Stories/WizardPatternExample.js`

**Comment by priley86** [comment]
ok i manually went through all of the storybooks earlier and couldn't find any issues. I'm good to go here once the WizardPatternExample and WizardExample are updated. 

can you please update rebase the commits to a single commit or just make the first commit a `fix()`... i.e. `fix(all): Remove bindMethods in favor of Class properties` ? This way we get those downstream I guess...(would like to validate they still work the same way).

**Comment by chalettu** [comment]
Hi @priley86 , I went ahead and took your advice and collapsed everything into one commit and also took care of the wizard that still had bindMethods.   Let me know if you have any questions

**Pull request merged**